pycbc/types/timeseries.py

```diff
@@ -84,12 +84,19 @@ def __init__(self, initial_array, delta_t=None, epoch="", dtype=None, copy=True)
         self._delta_t = delta_t
         self._epoch = epoch
 
+    def sample_rate_close(self, other):
+        """ Check if the sample rate is close enough to allow operations """
+        if abs(1 - other.delta_t / self.delta_t) * len(self) > 0.5:
```

Comment:
This seems like an odd condition, which might allow very short stretches of data to be multiplied with very different sample rates, but very long data must be almost exact.

Isn't the problem here that we want the two `delta_t` to actually be equal, but machine precision means that doesn't happen.

Reply:
Yes, though I'm not sure how one would decide exactly if they are close enough. i.e. it's a little arbitrary what the threshold would be as the values may have several times floating (or double) error depending on how they diverged. 

In practice maybe a fudge factor of a few * floating_epsilon would be sufficient?

Reply:
Maybe if the ratio of the two delta_t values is within like 1E-4 we can allow this? Although of course the problem is that this will shift the absolute time values of one of the two series, probably by a non-negligible amount for long series.

Reply:
@spxiwh That line of thinking is what led me to this metric, since it guarantees the end times couldn't be confused by a sample (i.e. it's unambiguous still that each sample has a best correspondence in the other vector). If speed wasn't a consideration, maybe it would make sense to impose both constrains? i.e. a fixed relative value along with the one I propose here?

Reply:
If *both* of these considerations are satisfied, then I don't see a problem with this. In practice there may not be so many cases where this condition comes up, but I'd have no issue with including it.